### ISSUE TITLE
Improvements to Describe > Graph > Scatterplot

### ISSUE THREAD

**Issue by rdstern** [IDEMSInternational/R-Instat#8800]
@derekagorhom this is another task mainly for @Raymond-Andam .  I hope it can be done in parallel with his other work.  I am keen for developers to have multiple tasks, because they may have to wait for help, or testing on each one.

This should initially be easier than some of the tasks on #8789.

Here is the current scatterplot dialog:

![image](https://github.com/IDEMSInternational/R-Instat/assets/11226469/72c35488-6abd-40d9-a6af-3e90d9768694)

a) Could you make a group box with label `Add.`  That is round the current `Add Line of Best Fit`, and `Add Rug Plot`  
b) In each of those please delete the word Add.  It doesn't have to be repeated.
c) Add another checkbox with label `Side Plot`.  For now it can be disabled.
d) Decrease the vertical space slightly between the 3 options in the group box labelled Geoms.  Also delete the s, so it is Geom.
e) Make that group box wider.  Then, in the Point row, add a checkbox before Size and another before Shape.  Default is unchecked, and then the up-down and drop down controls are hidden.
f) You don't need checkboxes for the other two, but could move theor controls over, so they line up neatly with those for the point option.
g) Change the R code generated slightly.  I think it now gives you the size, etc inside the geom_point(), and they should now only be when the checkbox is ticked.  (The reason is that it is possible also to have a size and/or a shape aesthetic (i.e. a variable).  If size and/or shape are given as a single value inside geom_point, then it will ignore the aesthetic.

That's all for you now, but I have a question for @lilyclements .

I am very keen to be able to add marginal plots to the scatterplots. 

![image](https://github.com/IDEMSInternational/R-Instat/assets/11226469/4693fae9-b14f-4f91-83c8-bc2d4e31bf6e)

I think they are useful anyway and partly for teaching.  The popular way most people seem to use is with ggExtra, asa shown above.  It is simple and nice and powerful, but has 2 defects to me anyway.
a) It is a slightly different syntax.
b) It seems not to cope with facetted plots.
A more recent package is ggside.  It will be a bit more complicated to add to the dialog, but I think it looks great.  Do you agree?

**Comment by Raymond-Andam** [comment]
@rdstern This partly fixes issue #8800

**Comment by lilyclements** [comment]
@rdstern only took me 8.5 months, not 5 years.
I do agree. I don't like the "wrapper" syntax in `ggExtras`.
`ggside` is consistent with the `ggplot2` syntax, and so I definitely agree we should use that option.
(Note that ggplot2 version >= 3.5.0 is required for 'ggside' to work).

**Comment by rdstern** [comment]
@fran2or I know you have looked at adding the ggside package already.  Could you look again and this time together with the geomtextpath package!

But this time could you try in a new way.  Eash of these packages adds a neat feature to an existing graph.  And each of them adds a lot of new geoms.  And that's no fun for our graph sub-dialog.

Here are the contents for ggside:
![image](https://github.com/user-attachments/assets/22741a10-9080-41dd-a8f0-72115659cfb6)

And here they are for ggtextpath

![image](https://github.com/user-attachments/assets/6f303bae-7e21-4e3a-b310-01a784d4b345)

Of course it is easy to add the packages into R-Instat.  So I now suggest we add thyem into R-Instat scripts instead!  So we get a graph (mainly scatterplot if going to add ggside) and mainly line plot (maybe smoothed) if ggtextpath.  Then get the script (using To script button) and then add the side plot or the textpath into the script.

Could you try to make a few sample scripts?  They could go into our new script library (just as complete scripts) and we describe how users could do this themselves, starting from the dialog.

If it works well, and there are sample additions that are common, we could later have a graph tab on our `insert` button in the log/script window.  

So, instead of adding these features to our dialogs, we make them easy to add as a payoff for those R-Instat users who are willing to learn a bit about R-Instat scripts.

I wonder if you have time this week to start looking at this?  It could become a nice component of an advanced R-Instat workshop.

**Comment by rdstern** [comment]
@derekagorhom one aspect that needs to change this year, is silence in replying to proposed issues.  This also applies to issues where I mention @fran2or because he still claims he does not receive messages in github that mention him.  So I expect you at least to comment here that you have passed an issue to him.  Sort of:

Just to confirm that I told @fran2or that github issue #8800 on side plots, etc needs his inputs and response.  I assume @fran2or has no problem using github - it is just that he is not alerted.

**Comment by derekagorhom** [comment]
@rdstern I did send @fran2or your comment and issue to him via mail.

**Comment by rdstern** [comment]
@derekagorhom thanks.  And please, in the future, not quietly, but report in the thread that this has been done.  Of course it would be so much better if @fran2or fixed his machine to receive them directly.

**Comment by fran2or** [comment]
@rdstern Below are the plots and scripts generated using the R-Instat To Script dialogue. Hope these are what are required?

 **- Scatter plot with ggside density plots**

![Image](https://github.com/user-attachments/assets/2dd2bdf4-021a-4cd6-83e2-dcf965fca791)


```
# Code generated by the dialog, Scatter Plot

mpg <- data_book$get_data_frame(data_name="mpg")
last_graph <- ggplot2::ggplot(data=mpg, mapping=ggplot2::aes(x=displ, y=hwy)) + ggplot2::geom_point() + theme_minimal()+ #adding ggside plots
  ggside::geom_xsidedensity(data=mpg, mapping=ggplot2::aes(y = after_stat(density)), fill = "lightblue", alpha = 0.5) +
  ggside::geom_ysidedensity(data=mpg, mapping=ggplot2::aes(x = after_stat(density)), fill = "lightgreen", alpha = 0.5)
data_book$add_object(data_name="mpg", object_name="last_graph", object_type_label="graph", object_format="image", object=check_graph(graph_object=last_graph))
data_book$get_object_data(data_name="mpg", object_name="last_graph", as_file=TRUE)
rm(list=c("last_graph", "mpg"))


```
 **- Line plot with geomtextpath**

![Image](https://github.com/user-attachments/assets/091ffb1b-888c-4217-9412-332170d69657)

```
# Code generated by the dialog, Line Plot

dodoma_by_year <- data_book$get_data_frame(data_name="dodoma_by_year", stack_data=TRUE, measure.vars=c("mean_tmin","mean_tmax"))
last_graph <- ggplot2::ggplot(data=dodoma_by_year, mapping=ggplot2::aes(y=value, colour=variable, x=year)) + ggplot2::geom_line(size=0.8) + theme_minimal()+ 
#inserting geomtextpath 
geomtextpath::geom_textpath(data=dodoma_by_year, mapping=ggplot2::aes(label =variable), hjust = 0.5, size = 6)
data_book$add_object(data_name="dodoma_by_year", object_name="last_graph", object_type_label="graph", object_format="image", object=check_graph(graph_object=last_graph))
data_book$get_object_data(data_name="dodoma_by_year", object_name="last_graph", as_file=TRUE)
rm(list=c("last_graph", "dodoma_by_year"))

```
 **- Combination Plot (ggside & geomtextpath)**

![Image](https://github.com/user-attachments/assets/ec6295c0-c935-4d5f-a3f6-6fa979dc64eb)

```
# Code generated by the dialog, Scatter Plot

mpg <- data_book$get_data_frame(data_name="mpg")
last_graph <- ggplot2::ggplot(data=mpg, mapping=ggplot2::aes(x=displ, y=hwy)) + ggplot2::geom_point() + #fiting smooth line using loess method
ggplot2::geom_smooth(method = "loess", color = "red", se = FALSE) + theme_minimal() + # Add side density plots using ggside
ggside::geom_xsidedensity(aes(y = after_stat(density)), fill = "lightblue", alpha = 0.5) + ggside::geom_ysidedensity(aes(x = after_stat(density)), fill = "lightgreen", alpha = 0.5) + # Add a labeled smooth line using geomtextpath
geomtextpath::geom_labelsmooth(aes(label = "Smooth Trend Line"),method = "loess",  color = "red", hjust = 0.5,size = 6)
data_book$add_object(data_name="mpg", object_name="last_graph", object_type_label="graph", object_format="image", object=check_graph(graph_object=last_graph))
data_book$get_object_data(data_name="mpg", object_name="last_graph", as_file=TRUE)
rm(list=c("last_graph", "mpg"))

```

**Comment by rdstern** [comment]
@fran2or that's ok as a start. What do you plan to do next?
For comparison you might like to look at Lily's bit of work on issue #8683.  It sounds different, but also leads to a new tab for the Insert dialog.  I fould what she did was perfect, while yours still waits for next steps in 

a) suggesting suitable data sets for illustration
b) Providing complete potential scripts
c) Considering how non-programmers could be helped, via the Insert dialog.

Looking forward.  Happy to chat if you would find that useful.

**Comment by fran2or** [comment]
@derekagorhom attached is the **insert dialogue design concept for the graphs** as discussed to start design implementation.

[R-Instat Graph.pptx](https://github.com/user-attachments/files/18732715/R-Instat.Graph.pptx)

**Comment by lilyclements** [comment]
@fran2or would you and @Ag-Derek be happy to take charge of improving the dialog in this issue?  My thought was:
 
* @fran2or to lead on the statistical / design side (it's pretty much all sorted in this issue, so simply just taking charge of the testing of it in R-Instat)
* @Ag-Derek  to handle the VB implementation side

**Comment by fran2or** [comment]
### Numerical vs Categorical

```R
survey <- data_book$get_data_frame(data_name="survey", stack_data=TRUE, measure.vars="yield")
last_graph <- ggplot2::ggplot(data=survey, mapping=ggplot2::aes(x=variety, y=value, fill=fertgrp)) +ggplot2::geom_boxplot() +  ggside::geom_ysideboxplot(mapping=ggplot2::aes(x=1, fill=fertgrp), alpha=0.35, orientation="x") + ggplot2::facet_wrap(facets=~variable, scales="free_x") + ggplot2::theme(ggside.panel.scale = 0.3, axis.text.x.top = ggplot2::element_blank(), axis.ticks.x.top = ggplot2::element_blank(), axis.title.x.top = ggplot2::element_blank())
data_book$add_object(data_name="survey", object_name="last_graph", object_type_label="graph", object_format="image", object=instatExtras::check_graph(graph_object=last_graph))
data_book$get_object_data(data_name="survey", object_name="last_graph", as_file=TRUE)
```

### Numerical vs Numerical

```R
survey <- data_book$get_data_frame(data_name="survey", stack_data=TRUE, measure.vars="yield")
last_graph <- ggplot2::ggplot(mapping=ggplot2::aes(x=size, y=value), data=survey) + ggplot2::geom_point(ggplot2::aes(colour=variety)) + ggside::geom_xsidedensity(alpha=0.35, ggplot2::aes(fill=variety), colour="NA") + ggside::geom_ysidedensity(alpha=0.35, ggplot2::aes(fill=variety), colour="NA") + ggplot2::facet_wrap(facets=~variable)+ ggplot2::theme(ggside.panel.scale = 0.3, axis.text.x.top = ggplot2::element_blank(), axis.ticks.x.top = ggplot2::element_blank(), axis.title.x.top = ggplot2::element_blank())
data_book$add_object(data_name="survey", object_name="last_graph", object_type_label="graph", object_format="image", object=instatExtras::check_graph(graph_object=last_graph))
data_book$get_object_data(data_name="survey", object_name="last_graph", as_file=TRUE)
rm(list=c("last_graph", "survey"))
```

**Comment by lilyclements** [comment]
@fran2or nice! What are the next steps here? Looks good r-code-wise
